### PR TITLE
Update Azure config with new template features

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -14,6 +8,7 @@ schedules:
       - master
       - 1.0
       - 1.1
```
Comment (nabobalis) [inline azure-pipelines.yml:10]:
```suggestion
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -67,8 +62,13 @@ jobs:
       - linux: py38-online
         name: py38_test_online
 
-      - linux32: py38-32bit
+      - manylinux: py38-32bit
         name: py38_test_32bit
+        manylinux_image: manylinux1_i686
```
Comment (nabobalis) [inline azure-pipelines.yml:67]: Can we not use an update to date image?

Comment (Cadair) [reply azure-pipelines.yml:67]: I think more of our deps provide wheels for this, which should make it a lot faster, but we should double check that.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -90,23 +84,24 @@ jobs:
...
+    # Only Upload to PyPI on tags
+    ${{ if startsWith(variables['Build.SourceBranch'], 'refs/tags/') }}:
+      pypi_connection_name : 'PyPI'
+    test_extras: 'dev'
+    test_command: 'pytest -p no:warnings --doctest-rst -m "not figure" --pyargs sunpy'
+    submodules: false
+
+    targets:
+      - wheels_cp[678]-manylinux*
+      - wheels_cp[678]-macosx*
```
Comment (nabobalis) [inline azure-pipelines.yml:99]:
```suggestion
      - wheels_cp[78]-manylinux*
      - wheels_cp[78]-macosx*
```
I assume assuming.

Comment (Cadair) [reply azure-pipelines.yml:99]: true but not for backports?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -1,19 +1,12 @@
 name: $(BuildDefinitionName)_$(Date:yyyyMMdd)$(Rev:.rr)
-variables:
-  CI_NAME: Azure Pipelines
-  CI_BUILD_ID: $(Build.BuildId)
-  CI_BUILD_URL: "https://dev.azure.com/sunpy/sunpy/_build/results?buildId=$(Build.BuildId)"
```
Comment (Cadair) [inline azure-pipelines.yml:5]: These aren't needed anymore because codecov now supports pipelines.

File: azure-pipelines.yml (hunk 3 above)
Comment (nabobalis) [reply azure-pipelines.yml]: We can add it in the backport?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -22,7 +15,7 @@ resources:
     type: github
     endpoint: sunpy
     name: OpenAstronomy/azure-pipelines-templates
-    ref: master
+    ref: manylinux_image
```
Comment (nabobalis) [inline azure-pipelines.yml:18]:
```suggestion
    ref: master
```

File: azure-pipelines.yml [hunk 6]
```diff
@@ -90,23 +83,24 @@ jobs:
...
+      - wheels_cp3[78]-manylinux*
+      - wheels_cp3[78]-macosx*
+      - sdist
+    #dependsOn:
+    #  - py37_test
+    #  - py38_test
+    #  - py38_test_online
+    #  - py38_test_32bit
+    #  - py37_test_oldestdeps
+    #  - py38_conda
```
Comment (nabobalis) [inline azure-pipelines.yml:106]:
```suggestion
    dependsOn:
      - py37_test
      - py38_test
      - py38_test_online
      - py38_test_32bit
      - py37_test_oldestdeps
      - py38_conda
```

Comment (nabobalis) [inline azure-pipelines.yml:106]: This entire block needs uncommeting, retabbing and the like.

File: azure-pipelines.yml [hunk 7]
```diff
@@ -90,23 +83,24 @@ jobs:
...
-        - wheels_macos
-        - sdist
-      dependsOn:
-        - py37_test
-        - py38_test
-        - py38_test_online
-        - py38_test_32bit
-        - py37_test_oldestdeps
-        - py38_conda
+# - ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), not(contains(variables['Build.SourceBranch'], 'master'))) }}:
```
Comment (Cadair) [inline azure-pipelines.yml:86]: needs uncommenting once the build passes

Comment (nabobalis) [inline azure-pipelines.yml:86]: Just to test it?

**Review by nabobalis — APPROVED** [review]

**Comment by Cadair** [comment]
Note this needs modifying on backport to enable py36 wheels.

**Comment by nabobalis** [comment]
> Note this needs modifying on backport to enable py36 wheels.

Can you add the suggestion?

**Comment by nabobalis** [comment]
Online build failed with:
```
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/_pytest/main.py", line 240, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/_pytest/main.py", line 296, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/xdist/dsession.py", line 112, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/xdist/dsession.py", line 135, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/xdist/dsession.py", line 174, in worker_workerfinished
INTERNALERROR>     assert not crashitem, (crashitem, node)
INTERNALERROR> AssertionError: ('.tox/py38-online/lib/python3.8/site-packages/sunpy/data/tests/test_sample.py::test_retry_sample_data', <WorkerController gw3>)
INTERNALERROR> assert not '.tox/py38-online/lib/python3.8/site-packages/sunpy/data/tests/test_sample.py::test_retry_sample_data'
```

I wonder if its worth just deleting the test.

File: azure-pipelines.yml [hunk 8]
```diff
@@ -99,9 +92,10 @@ jobs:
       test_extras: 'dev'
       test_command: 'pytest -p no:warnings --doctest-rst -m "not figure" --pyargs sunpy'
       submodules: false
+
       targets:
-        - wheels_linux
-        - wheels_macos
+        - wheels_cp3[78]-manylinux*x86_64
+        - wheels_cp3[78]-macosx*
```
Comment (Cadair) [inline azure-pipelines.yml:98]: For the backpotr:

```suggestion
        - wheels_cp3[678]-manylinux*x86_64
        - wheels_cp3[678]-macosx*
```

**Pull request merged**